### DISCUSSION TITLE
Enable hidden _system interfaces for custom use

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#15317] [Archive]
Right now the app uses a few custom interfaces that are not available for custom usage. 
Are there any special reasons why they're not enabled for custom usage?

I could imagine using some of them them for other use-cases like the [system-field-tree](https://github.com/directus/directus/tree/92e1ee77bd34118dce69e91c44d9cfcfbd208f37/app/src/interfaces/_system/system-field-tree) to store some custom field-related meta-data 🤔

**Comment by rijkvanzanten** [comment]
> Are there any special reasons why they're not enabled for custom usage?

Mostly because they're made to work in very specific contexts, and often don't rely on any configuration (but rather on page contexts/system integrations). 

Down to discuss "opening up" ones on a per-case basis, but I don't think we can (nor should) make every system one available by default 🙂

**Reply by Dominic-Marcelino** [reply]
Thanks! ☺️ 
Haven't checked all of them, but as mentioned I can imagine some use-cases for the `system-field-tree` as for example disabling fields on an entry level based on the tree-selection. 

But maybe it would first need an option like `first-level only` as you can't pass the values down to e.g to relational / translation interfaces. 🤔


On the other hand the most functionality happens inside the utils (calculating the tree). Maybe it would be the better solution to move those into shared and therewith provide this basic functionality for custom integrations? What would you think about that?

**Reply by rijkvanzanten** [reply]
Down for that! The only reason we're not exposing more internals in shared utils by default is mostly because we don't want to limit ourselves in the changes / fixes we can make in there 🙂 Anything we explicitly expose to end users for extension use needs to be tracked for breaking changes, which makes it a little more tricky to move fast sometimes
